twisted/internet/defer.py

```diff
@@ -661,6 +661,66 @@ def __str__(self):
     __repr__ = __str__
 
 
+    def __await__(self):
+        awaiter = _MakeDeferredAwaitable(self)
```

Comment:
Yeah, that's the right thing to do. Again, it's a micro-optimization that might or might not be worth implementing.